app/services/credentials.js

```diff
@@ -13,64 +18,64 @@ export default Service.extend({
   currentUser: service(),
   store: service(),
 
-  memberships: computed.alias('organization.organizationMemberships'),
-
-  membership: computed('memberships.@each.isNew', function() {
-    let memberships = this.get('_loadedMemberships');
-    if (memberships) {
-      return memberships.find((membership) => this.findMembership(membership));
+  membership: computed('memberships', 'user', {
+    get() {
+      get(this, 'membershipPromise').then((membership) => {
+        set(this, 'membership', membership);
+      });
+      return null;
+    },
+    set(k, val) {
+      return val;
     }
   }),
 
-  membershipPromise: computed('memberships', function() {
-    let memberships = this.get('memberships');
-    let fulfilled = this.get('memberships.isFulfilled');
-
+  // Memberships may not be loaded, so we need to wait for them to resolve
+  // Can be useful in routing when we need it loaded to prevent a transition
+  membershipPromise: computed('memberships', 'user', function() {
```

Comment:
It feels like this should just be a function like `fetchMembership` instead of a computer property. Since the `membership` property is already computed based on the same values, it should still be fine